src/site/shared/containers/HistoryBox/index.tsx

```diff
@@ -17,23 +17,42 @@ type HistoryEntryProps = {
     a: Action;
 }
 const HistoryEntry = ({ a }: HistoryEntryProps) => {
+    const [displayExtraInfo, setDisplayExtraInfo] = useState(true);
+
     if (a instanceof GroupAction)
         return (<GroupActionEntry g={a}></GroupActionEntry>);
     return (
         <div className="historybox__entry"
-             // Necessary to stop child entries from collapsing the parent history entry
-             onClick={(e) => e.stopPropagation()}>
-            {a.getName()}
+            onClick={(e) => {
+                // Necessary to stop child entries from collapsing the parent history entry
+                e.stopPropagation();
+                setDisplayExtraInfo(!displayExtraInfo);
+            }}>
+            <div className="historybox__entry__header">
+                {a.getCustomInfo &&
+                    <img className="historybox__entry__extrainfo__icon" src="img/icons/info.svg" alt="Display extra info"/>
+                }
+                <span className="historybox__entry__actionname">
+                    {a.getName()}
+                </span>
+            </div>
+            {!displayExtraInfo && a.getCustomInfo &&
+                a.getCustomInfo()?.map((obj, i) => {
+                    return (
+                        <div key={`entry-extrainfo-${i}`} className="historybox__entry__extrainfo">{obj}</div>
+                    )
+                })
+            }
```

Comment:
you can condense this to:
```tsx
{!displayExtraInfo && a.getCustomInfo()?.map((obj, i) =>
    <div ..... >{obj}</div>
)}
```

Reply:
If I don't include `a.getCustomInfo` in the condition, I get the error `Cannot invoke an object which is possibly 'undefined'.`.

Reply:
However, I am a little unsure why because I did not need this check in the below condition.

Reply:
Sorry for responding so late, for the first one, I lied -- it should've been `a.getCustomInfo?.()?.map(...)` which works correctly and the reason for the removing the `return` not working is presumably because the curly braces were still left in when they should also be deleted